File: CrackingTheCodingInterview/Chapter1/one-seven.py
# Rotate a matrix 90 clockwise in place
import math
def rotate(matrix):
    temp = 0
    size = len(matrix)
    offset = size-1
    for x in range(0, math.ceil(size/2)):
        for y in range(0, math.ceil(size/2)):
            temp = matrix[x][y]
            matrix[x][y] = matrix[offset-y][x]
            matrix[offset-y][x] = matrix[offset-x][offset-y]
            matrix[offset-x][offset-y] = matrix[offset-x][y]
            matrix[offset-x][y]=temp
    return matrix

File: CrackingTheCodingInterview/Chapter1/OneSeven.java
// rotate a matrix 90 degrees clockwise in place
public class OneSeven 
{
    static int[][] rotate(int[][] array)
    {
        int size = array.length;
        int temp = 0;
        int offset = size-1;
        for (int x = 0; x < size/2; x++)
        {
            for (int y = 0; y < size/2; y++)
            {
                temp = array[x][y];
                array[x][y] = array[offset-y][x];
                array[offset-y][x] = array[offset-x][offset-y];
                array[offset-x][offset-y] = array[offset-x][y];
                array[offset-x][y] = temp;
            }
        }
        return array;
    }
    
}